### PR TITLE
NPE fix: action can be null in handleImageBuildData bsc#1185951

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?
NPE is:
```
java.lang.NullPointerException
	at com.suse.manager.utils.SaltUtils.handleImageBuildData(SaltUtils.java:1149)
	at com.suse.manager.utils.SaltUtils.updateServerAction(SaltUtils.java:622)
	at com.suse.manager.webui.services.SaltServerActionService.lambda$handleAction$159(SaltServerActionService.java:2722)
```
Since the NPE happens also when everything is working fine, let' s put the root cause NPE variable as optional (but let's create a  warning if it's null).

Moreover, unproxy function is removed as suggested in the comment:
```
Impelmentation is copied from Hibernate 5.2.10 and this method should be removed after that release.
```
## GUI diff

No difference.

- [ ] **DONE**

## Documentation
- No documentation needed

- [ ] **DONE**

## Test coverage

- [ ] **DONE**

## Links

https://bugzilla.suse.com/show_bug.cgi?id=1185951
https://github.com/SUSE/spacewalk/issues/14927

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [x] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql" (Test skipped, there are no changes to test)
- [ ] Re-run test "java_pgsql_tests" 
- [x] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql" (Test skipped, there are no changes to test)
- [ ] Re-run test "susemanager_unittests" (Test skipped, there are no changes to test)
- [x] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests" (Test skipped, there are no changes to test)

### CODE REVIEW

File: java/code/src/com/suse/manager/utils/SaltUtils.java [hunk 1]
```diff
@@ -1133,67 +1133,70 @@ private void handleScapXccdfEval(ServerAction serverAction,
...
-                LOG.error(e);
             }
-
-            infoOpt.ifPresent(info -> {
-                info.setRevisionNumber(info.getRevisionNumber() + 1);
-                info.setInspectAction(iAction);
-                ImageInfoFactory.save(info);
-            });
-        }
+        }, () -> LOG.warn("ServerAction does not have any parent "));
```
Comment (mcalmer) [inline java/code/src/com/suse/manager/utils/SaltUtils.java:1199]: Should we be a bit more verbose? Can we add more information to help somebody to understand and find the root cause of this when he read this message in the log?
Is a "warning" the right thing here? Why is this not an error?

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: java/code/src/com/suse/manager/utils/SaltUtils.java (hunk 1 above)
Comment (mbussolotto) [reply java/code/src/com/suse/manager/utils/SaltUtils.java:1199]: Totally right, moreover I found out that I was not takling the wrong NPE

**Comment by mbussolotto** [comment]
@mcalmer , since this NPE happens only during the secondary features of acceptance test, is there any way for testing this PR again that test? I used "uyuni-prs-ci-tests" in the past, but it seems it doesn't include secondary features.

**Comment by mbussolotto** [comment]
trying to run acceptance test to check if the logs looks better with these changes (the NPE happens only during secondary features tests)

**Review by aaannz — APPROVED** [review]
LGTM

**Comment by mcalmer** [comment]
@mbussolotto you can ask @jordimassaguerpla how to run the acceptance tests with only smaller feature set.

**Review by mcalmer — APPROVED** [review]

**Comment by mbussolotto** [comment]
waiting for https://github.com/SUSE/susemanager-ci/pull/349 to fix current issue in acceptance test pipeline. Then I will test my changes using this pipeline

**Pull request merged**